programs/gumball-machine/src/lib.rs

```diff
@@ -284,87 +433,90 @@ pub mod gumball_machine {
             None => {}
         }
         match bot_wallet {
+            // TODO(sorend): do we want to apply some validation here?
+            // If this will only collect SOL then this is fine, if not then need to validate that it's a TokenAccount with a mint corresponding to project mint
```

Comment:
add validation function + test

test should pass in a non-token account as the wallet & initialization should fail :)

Reply:
@jarry-xiao and I talked abt this briefly today. I think we can skip validation here because the `bot_wallet` is only going to receive SOL. 

But we might need to validate the `receiver` (the address we payout mint fees to) passed to initialize_gumball_machine. @ngundotra do you have thoughts on this? I think the current options @jarry-xiao talked about are:
a) IF `mint` is not native -> check that the `Pubkey` of the `receiver` is the associated token address for `mint` with owner `authority.key()`.
b) Pass `receiver` as an account in `ctx`, check that we can unpack the `receiver` account into some TokenAccount with mint: `mint`
c) Validate nothing, user's fault if they init broken machine

Referring to this: https://github.com/jarry-xiao/candyland/blob/566cd5ae3ce596f29585c5317f0a55cd976b0f40/programs/gumball-machine/src/lib.rs#L263

Reply:
ah I see. yeah seems fine to skip validation. bots should be taxed in SOL